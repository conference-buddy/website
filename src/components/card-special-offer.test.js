import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import CardSpecialOffer from "./card-special-offer";

const cardTitle = "A card title";
const requiredProps = {
  title: cardTitle,
  subtitle: "subtitle",
  children: "<div>hello</div>",
  disclaimerText: "disclaimer text",
};
describe("<CardSpecialOffer />", () => {
  let component;
  beforeAll(() => {
    component = render(<CardSpecialOffer {...requiredProps} />);
  });

  afterAll(cleanup);

  it("shows a header for the card", () => {
    const header = screen.getByRole("heading", {
      name: `ðŸ”¥ ${cardTitle} ðŸ”¥`,
    });

    expect(header).toBeVisible();
  });

  it("shows all elements", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
