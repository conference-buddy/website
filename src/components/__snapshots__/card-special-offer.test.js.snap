// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSpecialOffer /> shows all elements 1`] = `
<DocumentFragment>
  <div
    class="card bg-light shadow-sm m-md-2"
  >
    <div
      class="card-body"
    >
      <h3
        class="card-title text-secondary text-center pb-2"
      >
        ðŸ”¥ A card title ðŸ”¥
      </h3>
      <h6
        class="card-subtitle mb-2 text-muted "
      >
        subtitle
      </h6>
      &lt;div&gt;hello&lt;/div&gt;
    </div>
    <div
      class="card-footer text-muted small"
    >
      <span
        class="text-secondary"
      >
        (*) Please note:
      </span>
       disclaimer text
    </div>
  </div>
</DocumentFragment>
`;
